lib/main/atom/components/miniEditor.ts

```diff
@@ -0,0 +1,65 @@
+import {TextEditor, TextEditorElement} from "atom"
+
+interface Props extends JSX.Props {
+  initialText: string
+  selectAll?: boolean
+  readOnly?: boolean
+  grammar?: string
+}
+
+export class MiniEditor {
+  private model: TextEditor
+  public element: TextEditorElement
+
+  constructor(public props: Props) {
+    this.model = atom.workspace.buildTextEditor({
+      mini: true,
+      softWrapped: true,
+      lineNumberGutterVisible: false,
+    })
+    this.element = atom.views.getView(this.model)
+    this.model.setText(props.initialText)
+    if (props.selectAll) {
+      this.model.selectAll()
+    } else {
+      this.model.getLastCursor().moveToEndOfScreenLine()
+    }
+    if (props.readOnly) {
+      this.element.removeAttribute("tabindex") // make read-only
+    }
+    if (props.grammar) {
+      const grammar = atom.grammars.grammarForScopeName(props.grammar)
+      if (grammar) {
+        this.model.setGrammar(grammar)
+      }
+    }
+    this.model.scrollToBufferPosition([0, 0])
+  }
+
+  public async update(props: Props) {
+    this.element = atom.views.getView(this.model)
+    if (this.props.readOnly !== props.readOnly) {
```

Comment:
TODO: check for undefined in `props.readOnly`, otherwise partial update is impossible